src/main/java/at/hannibal2/skyhanni/features/garden/visitor/VisitorRewardWarning.kt

```diff
@@ -0,0 +1,128 @@
+        if (!config.preventRefusing && !config.preventRefusingCopper && !config.preventAcceptingCopper) return
+
+        val visitor = VisitorAPI.getVisitor(lastClickedNpc) ?: return
+        val blockReason = visitor.blockReason(true, true, true)
+        val refuseOfferStack = event.gui.inventorySlots.getSlot(REFUSE_SLOT).stack ?: return
+        val acceptOfferStack = event.gui.inventorySlots.getSlot(ACCEPT_SLOT).stack ?: return
+
+        if (blockReason == VisitorBlockReason.RARE_REWARD || blockReason == VisitorBlockReason.CHEAP_COPPER) {
+            if (!config.bypassKey.isKeyHeld()) refuseOfferStack.background = LorenzColor.DARK_GRAY.addOpacity(config.opacity).rgb
+            if (config.optionOutline) acceptOfferStack.borderLine = LorenzColor.GREEN.addOpacity(200).rgb
+            return
+        }
+        if (blockReason == VisitorBlockReason.EXPENSIVE_COPPER) {
+            if (!config.bypassKey.isKeyHeld()) acceptOfferStack.background = LorenzColor.DARK_GRAY.addOpacity(config.opacity).rgb
+            if (config.optionOutline) refuseOfferStack.borderLine = LorenzColor.RED.addOpacity(200).rgb
+            return
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    fun onStackClick(event: SlotClickEvent) {
+        if (!VisitorAPI.inInventory) return
+
+        val visitor = VisitorAPI.getVisitor(lastClickedNpc) ?: return
+        val blockReason = visitor.blockReason(event.slotId == ACCEPT_SLOT, event.slotId == REFUSE_SLOT)
+
+        if (blockReason == VisitorBlockReason.RARE_REWARD || blockReason == VisitorBlockReason.CHEAP_COPPER) {
+            event.isCanceled = true
+            return
+        }
+
+        if (blockReason == VisitorBlockReason.EXPENSIVE_COPPER) {
+            event.isCanceled = true
+            return
+        }
+
+        if (event.slotId == REFUSE_SLOT) {
+            if (event.slot.stack?.name != "§cRefuse Offer") return
+            VisitorAPI.changeStatus(visitor, VisitorAPI.VisitorStatus.REFUSED, "refused")
+            return
+        }
+        if (event.slotId == ACCEPT_SLOT) {
+            if (event.slot.stack?.name != "§eClick to give!") return
+            VisitorAPI.changeStatus(visitor, VisitorAPI.VisitorStatus.ACCEPTED, "accepted")
+            return
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    fun onTooltip(event: ItemTooltipEvent) {
+        if (!GardenAPI.onBarnPlot) return
+        if (!VisitorAPI.inInventory) return
+        val visitor = VisitorAPI.getVisitor(lastClickedNpc) ?: return
+
+        val isRefuseSlot = event.itemStack.toString() == "1xtile.clayHardenedStained@14" // red
+        val isAcceptSlot = event.itemStack.toString() == "1xtile.clayHardenedStained@13" // green
+
+        val blockReason = visitor.blockReason(isAcceptSlot, isRefuseSlot)
+        if (blockReason == VisitorBlockReason.NONE) return
+
+        val copiedTooltip = event.toolTip.toList()
+        event.toolTip.clear()
```

Comment:
drawing the tooltip every frame is a bit overkill. can we prepare the list on inventory open and then use this cache here?